src/modules/dashboard/components/ExpenditurePage/Stages/Stages.tsx

```diff
@@ -70,19 +83,35 @@ interface Props {
   states: State[];
   handleSubmit: (values: InitialValuesType) => void;
   activeStateId?: string;
+  recipients?: RecipientType[];
+  colony?: Colony;
 }
 
-const Stages = ({ states, activeStateId }: Props) => {
-  const { values, handleSubmit, validateForm } =
+const Stages = ({ states, activeStateId, recipients, colony }: Props) => {
+  const { values, handleSubmit, validateForm, resetForm } =
     useFormikContext<ValuesType>() || {};
 
-  const { resetForm } = useFormikContext() || {};
   const [valueIsCopied, setValueIsCopied] = useState(false);
   const userFeedbackTimer = useRef<any>(null);
 
   const openDeleteDraftDialog = useDialog(DeleteDraftDialog);
   const openDraftConfirmDialog = useDialog(StakeExpenditureDialog);
 
+  const recipientsWithTokens = useMemo(() => {
+    return recipients?.map((recipient) => {
+      const token = getRecipientTokens(recipient, colony);
+      return { ...recipient, value: token };
+    });
+  }, [colony, recipients]);
+
+  const {
+    claimableNow,
+    claimed,
+    totalClaimable,
```

Comment:
I think that all of the about should be an array of objects if there are several tokens/amounts per recipient. Then you don't need to create arrays from them later.